ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -20,6 +20,55 @@ open D
 
 open Stdext.Xstringext
 
+let lwsmd_service = "lwsmd"
+
+module Lwsmd = struct
+
+  let is_ad_enabled ~__context =
+    Helpers.get_localhost ~__context
+    |> (fun self -> Db.Host.get_external_auth_type ~__context ~self)
+    |> (fun x -> x = Xapi_globs.auth_type_AD_Likewise)
+
+  let is_bootup() =
+    Fe_systemctl.is_active lwsmd_service
+
+  let is_stoped() =
+    not (is_bootup())
+
+  type t =
+    | Start
+    | Stop
+
+  let to_string = function
+    | Start -> "start"
+    | Stop -> "stop"
+
+  let operate ?(wait_until_success=false) op =
+    let op_str = op |> to_string in
+    try
+      ignore (Forkhelpers.execute_command_get_output !Xapi_globs.systemctl [op_str; lwsmd_service]);
+      let timeout = 2. in
+      if wait_until_success then
+        let success_cond = match op with
+          | Start -> is_bootup
+          | Stop -> is_stoped in
+        try
+          Helpers.retry_until_timeout ~timeout (Printf.sprintf "trying to %s %s" op_str lwsmd_service) success_cond
+        with _ -> debug "Fail to %s %s timeout %f" op_str lwsmd_service timeout
+    with e -> error "Fail to %s %s with error %s" op_str lwsmd_service (ExnHelper.string_of_exn e)
+
+  let stop ~wait_until_success =
+    operate ~wait_until_success Stop
+
+  let start ~wait_until_success =
+    operate ~wait_until_success Start
+
+  let init_service  ~__context =
+    (* This function is called during xapi start *)
+    (* it will start lwsmd service if the host is authed with AD *)
+    if is_ad_enabled ~__context then start ~wait_until_success:false
```

Comment:
I guess the code does not wait for the service to be active becase
1. We do not want to add more time to xapi boot process
2. We do not want xapi to fail if he external AD fails

Regardless of the reason it would be good to explain them, could you add a comment with the reasons for this?
I'm happy with the comment being in the commit message or here in the code.

Reply:
updated